libzerocash/Address.cpp

```diff
@@ -97,25 +116,28 @@ Address::Address(): addr_pk(), a_sk(a_sk_size) {
 
     privateKey.Save(StringSink(encodedPrivateKey).Ref());
 
-    this->sk_enc = encodedPrivateKey;
-
-    addr_pk.createPublicAddress(this->a_sk, this->sk_enc);
+    addr_pk.createPublicAddress(a_sk, encodedPrivateKey);
+    addr_sk.createPrivateAddress(a_sk, encodedPrivateKey);
 }
 
 const PublicAddress& Address::getPublicAddress() const {
 	return this->addr_pk;
 }
 
+const PrivateAddress& Address::getPrivateAddress() const {
+    return this->addr_sk;
+}
+
 const std::string Address::getEncryptionSecretKey() const {
-    return this->sk_enc;
+    return this->addr_sk.getEncryptionSecretKey();
 }
 
 const std::vector<unsigned char>& Address::getAddressSecret() const {
-    return this->a_sk;
+    return this->addr_sk.getAddressSecret();
 }
 
 bool Address::operator==(const Address& rhs) const {
```

Comment:
(Does not block PR.) Non-constant time comparison of secrets is a security vuln.